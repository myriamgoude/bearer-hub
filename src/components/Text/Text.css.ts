import { css } from 'emotion'
import { getEmSize } from '../../styles/mixins'
import { colors, dimensions } from '../../styles/variables'

export default {
  root: css`
    color: ${colors.branded.darkBlue};
    font-size: ${getEmSize(dimensions.fontSize.large)};
  `,
  fontSmall: css`
    font-size: ${getEmSize(dimensions.fontSize.small)};
  `,
  fontLarge: css`
    font-size: ${getEmSize(dimensions.fontSize.large)};
  `
}
