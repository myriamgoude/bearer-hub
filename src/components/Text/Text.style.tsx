import { css } from '@emotion/core'
import { getEmSize } from '../../styles/mixins'
import { colors, dimensions } from '../../styles/variables'

export default {
  root: css`
    display: inline-block;
    color: ${colors.darkBlue};
    font-size: ${getEmSize(dimensions.fontSize.large)};
  `,
  fontSmall: css`
    font-size: ${getEmSize(dimensions.fontSize.small)};
  `,
  fontLarge: css`
    font-size: ${getEmSize(dimensions.fontSize.large)};
  `
}
