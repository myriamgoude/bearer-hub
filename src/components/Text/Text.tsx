import React from 'react'
import styles from './Text.style'

interface ITextProps {
  text?: string | any
  className?: any
  small?: boolean
  large?: boolean
}

const Text = (props: ITextProps) => {
  return (
    <p
      css={[
        styles.root,
        props.className && props.className,
        props.small && styles.fontSmall,
        props.large && styles.fontLarge
      ]}
    >
      {props.text}
    </p>
  )
}

export default Text
