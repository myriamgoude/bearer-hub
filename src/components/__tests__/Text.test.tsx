import * as React from 'react'
import renderer from 'react-test-renderer'
import Text from '../Text/Text'

describe('Text', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Text text="hello" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with className', () => {
    const tree = renderer.create(<Text text="hello" style="foo" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
