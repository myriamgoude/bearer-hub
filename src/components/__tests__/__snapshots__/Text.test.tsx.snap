// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<p
  css={
    Array [
      Object {
        "map": undefined,
        "name": "1o5r80t",
        "next": undefined,
        "styles": "
    color: #030D36;
    font-size: 1.125rem;
  ",
      },
      undefined,
      undefined,
      undefined,
    ]
  }
>
  hello
</p>
`;

exports[`Header renders correctly with className 1`] = `
<p
  css={
    Array [
      Object {
        "map": undefined,
        "name": "1o5r80t",
        "next": undefined,
        "styles": "
    color: #030D36;
    font-size: 1.125rem;
  ",
      },
      "foo",
      undefined,
      undefined,
    ]
  }
>
  hello
</p>
`;
